components/db/db-connection-manager-impl/src/main/kotlin/net/corda/db/connection/manager/impl/DbConnectionManagerImpl.kt

```
@@ -74,6 +76,10 @@ class DbConnectionManagerImpl (
         lifecycleCoordinator.postEvent(BootstrapConfigProvided(config))
     }
 
+    override fun testAllConnections(): Boolean {
+        return dbConnectionsRepository?.testAllConnections() ?: false
```

As far as I can tell if dbConnectionsRepository is null this is a programming error as the scheduler should not be kicked off until it isn't. I agree that we should be defensive here, but I think it's worth logging an error rather than silently ignoring the code isn't working as we expect.

---

Done